spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SkipGramTrainer.java

```diff
@@ -0,0 +1,172 @@
+    model.save(outputFS.getBaseLocation().append(config.path).toURI().getPath());
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext sparkPluginContext) throws Exception {
+    //no-op
+  }
+
+  /**
+   * Configuration for SkipGramTrainer.
+   */
+  public static class SkipGramTrainerConfig extends PluginConfig {
+    @Description("The name of the FileSet to save the model to.")
+    private String fileSetName;
+
+    @Description("Path of the FileSet to save the model to.")
+    private String path;
+
+    @Description("Field to be used for training.")
+    private String inputCol;
+
+    @Nullable
+    @Description("The dimension of codes after transforming from words. Default 3.")
+    private Integer vectorSize;
+
+    @Nullable
+    @Description("The minimum number of times a token must appear to be included in the word2vec model's vocabulary. " +
+      "Default 0.")
+    private Integer minCount;
+
+    @Nullable
+    @Description("Sets number of partitions. Use a small number for accuracy. Default is 1.")
+    private Integer numPartitions;
+
+    @Nullable
+    @Description("Maximum number of iterations (>= 0). Default 1.")
+    private Integer numIterations;
+
+    @Description("The window size (context words from [-window, window]). Default 5.")
+    @Nullable
+    private Integer windowSize;
+
+    public SkipGramTrainerConfig() {
+      vectorSize = 3;
+      minCount = 0;
+      numPartitions = 1;
+      numIterations = 1;
+      windowSize = 5;
+    }
+
+    public SkipGramTrainerConfig(String fileSetName, String path, String inputCol, @Nullable int vectorSize,
+                                 @Nullable int minCount, @Nullable int numPartitions, @Nullable int numIterations,
+                                 @Nullable int windowSize) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.inputCol = inputCol;
+      this.vectorSize = vectorSize;
+      this.minCount = minCount;
+      this.numPartitions = numPartitions;
+      this.numIterations = numIterations;
+      this.windowSize = windowSize;
+    }
+
+    private void validate(Schema inputSchema) {
+      Schema.Field field = inputSchema.getField(inputCol);
+      if (field == null) {
+        throw new IllegalArgumentException(String.format("Input column %s does not exist in the input schema",
+                                                         inputCol));
+      }
+      Schema schema = field.getSchema();
+      Schema.Type type = schema.isNullable() ? schema.getNonNullable().getType() : schema.getType();
+      if (!(type.equals(Schema.Type.STRING) || type.equals(Schema.Type.BOOLEAN) || (type.equals(Schema.Type.ARRAY) &&
+        (schema.getComponentSchema().getType().equals(Schema.Type.STRING) ||
+          schema.getComponentSchema().getType().equals(Schema.Type.BOOLEAN))))) {
+        throw new IllegalArgumentException(String.format("Input column can be of type string, boolean or array of " +
```

Comment:
Agreed. Will remove the boolean types.
And regarding the array of strings:
The array of strings was kept considering scenarios where the user has already converted the data into tokens(array of string - maybe created using the tokenizer plugin), which can be directly used as an input for the text based feature generation.
Please let me know if we can have the field to be of type string, nullable string or array of strings or this needs to be changed to accept only string or nullable string.